### PR TITLE
[WIP] Add better `sample_weights` test coverage for `Metrics`

### LINKED ISSUE #16002: Better `sample_weights` test coverage for Metrics
It seems that in at least one case in a Metric object, the `sample_weight` argument wasn't being tested. See: #15997 #15939

We should add coverage for `sample_weight` in a systematic manner to the tests for all metrics. Right now it seems we have generic tests for sample weights, but not systematic tests for each specific metrics.

### PR DESCRIPTION
**Pull request by awinml**
#### Reference Issues/PRs
Fixes #16002 

#### What does this implement/fix? Explain your changes.
Systematically add tests for `sample_weights` argument in `metrics_test.py`.

### CODE REVIEW

**Comment by rchao** [comment]
Looks like there is only one line change in this PR which doesn't seem to intend to be final.

**Comment by awinml** [comment]
@rchao This PR is still a work in progress (WIP). I am still triaging the issue and will first add a complete list of metrics for which the tests need to be added or modified.
Once that is done I will push the required changes.

**Comment by gbaned** [comment]
Hi @rchao Can you please assist on above comments from @awinml. Thank you!

**Comment by rchao** [comment]
There doesn't seem anything actionable for Keras team at the moment.

**Comment by gbaned** [comment]
Hi @awinml Any update on this PR? Please. Thank you!

**Comment by awinml** [comment]
@gbaned @rchao As discussed in #16010 and #15997, I have added two tests similar to what was mentioned. I intend to add similar such tests for the other metrics as well.

Your suggestions would be welcome.

File: keras/metrics/metrics_test.py
Comment (rchao) [inline keras/metrics/metrics_test.py:1]: Thanks for the PR. Is it intended that this test is completely removed?

Comment (awinml) [reply keras/metrics/metrics_test.py:1]: Yes. When I started working on this PR the organization of the tests for all the metrics was only in a couple of files. So the initial plan was to create a new file for all the new tests. After merging with master yesterday, git pulled in all the tests from the other files for each category of metrics ie. `accuracy_metrics_test.py`, `confusion_metrics_test.py` etc into `metrics_test.py`.

After review, I realized the new tests can be added in their respective files, as compared to creating a completely new file.

For now I have added only two tests. Are those okay? Should I add similar tests like those for every metric?

Comment (rchao) [reply keras/metrics/metrics_test.py:1]: I believe in the [referenced issue](https://github.com/keras-team/keras/issues/16002), @fchollet was looking for a systematic way to cover the argument, and not individually.

Comment (awinml) [reply keras/metrics/metrics_test.py:1]: I am not sure how to go about doing that. Do you have any examples that I refer?

Comment (rchao) [reply keras/metrics/metrics_test.py:1]: We could probably utilize the combination api to test different sample weights, or create a decorator. Francois may also have some thoughts.

Comment (awinml) [reply keras/metrics/metrics_test.py:1]: @fchollet What approach should I follow for creating these tests?

Comment (awinml) [reply keras/metrics/metrics_test.py:1]: @rchao @fchollet Any suggestions on https://github.com/keras-team/keras/pull/17262#discussion_r1147451991?

Comment (gbaned) [reply keras/metrics/metrics_test.py:1]: Hi @rchao @fchollet Any update on this PR? Please. Thank you!

**Pull request closed without merging**